index.js

```diff
@@ -0,0 +1,37 @@
+const path = require('path');
+const { JsonFilter } = require('jsfilter');
+
+const mergeBits = ({ subject, bits }) => (
+  bits
+    .filter((bit) => {
+      // only bits which criteria matches the subject should remain
+      if (bit.criteria) {
+        try {        
+          return JsonFilter
+            .create(bit.criteria)
+            .match(subject);
+        }
+        catch (error) {
+          console.log('Failed criteria', bit.criteria, error);
+          return false;
+        }
+      }
+
+      return true;
+    })
+    .sort((a,b) => (a.priority || 0) - (b.priority || 0))
+    .reduce((res, bit) => Object.assign(res, bit.data), {})
+);
+
+module.exports = ({ bits = [] }) => {
+  return {
+    bits,
+    set(bits) {
+      this.bits = bits;
+    },
+    get(subject) {
```

Comment:
subject is kind of a weird name? This seems more like a query

Reply:
Was struggling to find a good name.
It is the object we apply the criteria to.

Does query fit that definition?

Reply:
You're querying the confbit object to give you a config that answers your query, is the way I was looking at it

Reply:
Sounds good, I will switch